controller/replica_controller.go

```diff
@@ -234,23 +214,24 @@ func (rc *ReplicaController) syncReplica(key string) (err error) {
 		return err
 	}
 
+	if replica.DeletionTimestamp != nil {
+		// delete instance, remove node storage and finally remove finalizer
+		if replica.Spec.NodeID == rc.controllerID {
+			if err := rc.instanceHandler.DeleteInstanceForObject(replica); err != nil {
+				return err
+			}
+			if err := os.RemoveAll(replica.Spec.DataPath); err != nil {
```

Comment:
You need to follow the logic in the original clean up logic. You will break live upgrade and migration in this way.